src/site/content/en/blog/when-to-use-local-https/index.md

```diff
@@ -0,0 +1,101 @@
+
+  {% Aside 'gotchas' %}
+  When it comes to setting `Secure` cookies locally, not all browsers behave in the same way! For example, Chrome and Safari don't set `Secure` cookies on localhost, but Firefox does. In Chrome, this is considered a bug.
+  {% endAside %}
+
+- You need to locally test or reproduce a behaviour specific to HTTP/2 or newer. For example, if you need to test loading performance on HTTP/2 or newer. Insecure HTTP/2 or newer is not supported, not even on `localhost`.
+- You need to locally test third-party libraries or APIs that require HTTPS (for example OAuth).
+- You're not using `localhost`, but a custom host name for local development, for example `mysite.example`. Typically, this means you've overridden your local hosts file:
+  <figure class="w-figure">
+    <img src="./hosts.jpg" alt="Screenshot of a terminal editing a hosts file">
+    <figcaption class="w-figcaption">Editing a hosts file to add a custom hostname.</figcaption>
+  </figure>
+
+  In this case, Chrome, Edge, Safari, and Firefox by default do _not_ consider `mysite.example` to be secure, even though it's a local site. So it won't behave like an HTTPS site.
+
+- Other cases! This is not an exhaustive list, but if you encounter a case that's not listed here, you'll know: things will break on `http://localhost`, or it won't quite behave like prod. 🙃
+
+**In all of these cases, you need to use HTTPS for local development.**
+
+## How to use HTTPS for local development
+
+If you need to use HTTPS for local development, head over to [How to use HTTPS for local development](/how-to-use-local-https).
+
+## Tips if you'e using a custom hostname
+
+**If you're using a custom hostname, i.e. if you're editing your hosts file:**
+
+- Don't use a bare hostname like `mysite`. Because if there's a [top-level domain (TLD)](https://en.wikipedia.org/wiki/Top-level_domain) that happens to have the same name (`mysite`), you'll run into issues. And it's not that unlikely: in 2020, there are over 1,500 TLDs, and the list is growing. `coffee`, `museum`, `travel`, and many large company names (maybe even the company you're working at!) are TLDs. [See the full list here](https://data.iana.org/TLD/tlds-alpha-by-domain.txt).
+- Only use domains that are yours, or that are reserved for this purpose. If you don't have a domain of your own, you can use either `test` or `localhost` (`mysite.localhost`). `test` doesn't have special treatment in browsers, but `localhost` does: Chrome and Edge support `http://<name>.localhost` out of the box, and it will behave securely when localhost does. Try it out: run any site on localhost and access `http://<whatever name you like>.localhost:<your port>` in Chrome or Edge. This may soon be possible in Firefox and [Safari](https://bugs.webkit.org/show_bug.cgi?id=160504) as well. The reason you can do this, i.e. have subdomains like `mysite.localhost`, is because `localhost` is not just a hostname: it's also a full TLD, like `com`.
```

Comment:
```suggestion
- Only use domains that are yours, or that are reserved for this purpose. If you don't have a domain of your own, you can use either `test` or `localhost` (`mysite.localhost`). `test` doesn't have special treatment in browsers, but `localhost` does: Chrome and Edge support `http://<name>.localhost` out of the box, and it will behave securely when localhost does. Try it out: run any site on localhost and access `http://<whatever name you like>.localhost:<your port>` in Chrome or Edge. This may soon be possible in Firefox and [Safari](https://bugs.webkit.org/show_bug.cgi?id=160504) as well. The reason you can do this (have subdomains like `mysite.localhost`) is because `localhost` is not just a hostname: it's also a full TLD, like `com`.
```